### ISSUE TITLE
[core] build problem: "The JAVA_HOME environment variable is not defined correctly" in pmd-apex build

### ISSUE THREAD

**Issue by Nazdravi** [pmd/pmd#859]
Please, prefix the report title with the language it applies to within brackets, such as [java] or [apex].
If not specific to a language, you can use [core].

**Affects PMD Version:**  6.0.0 and current git

Hi,
when building PMD with jre9-openjdk 9.u181-4 (Arch Linux) the process fails in pmd-apex, complaining about missing JAVA_HOME variable ([see log of build process](https://www.logicway.de/transfer/pmd.build.log)). Maven (also without specifying it with -D) has a valid java home setting (see below).

Regards
Nazdravi
```
$ mvn --version
Picked up _JAVA_OPTIONS: -Dhttp.proxyHost=proxy -Dhttp.proxyPort=2992 -Dhttps.proxyHost=proxy -Dhttps.proxyPort=2992
Apache Maven 3.5.2 (NON-CANONICAL_2017-10-25T13:09:52+03:00_root; 2017-10-25T12:09:52+02:00)
Maven home: /opt/maven
Java version: 9, vendor: Oracle Corporation
Java home: /usr/lib/jvm/java-9-openjdk
Default locale: de_DE, platform encoding: UTF-8
OS name: "linux", version: "4.14.13-1-arch", arch: "amd64", family: "unix"
```

**build command:** ./mvnw -Djava.home=/usr/lib/jvm/java-9-openjdk/ -Dpmd.skip=true clean verify

**Comment by adangel** [comment]
If the build really fails because of missing JAVA_HOME, then it's about the environment variable missing (and not the system property). The full command would be then:
`JAVA_HOME=/usr/lib/jvm/java-9-openjdk ./mvnw -Dpmd.skip=true clean verify`

The attached build log however fails with the module pmd-visualforce and according to the error message, it fails when executing the maven pmd plugin. This indicates, the build was executed without "-Dpmd.skip=true".

Background info:
The requirement for the JAVA_HOME environment variable and build problem due to java version will be fixed with #854 . The requirement itself was introduced via the fix in #842 .

The problem with pmd.skip=true will be fixed, as soon as PMD 6.0.1 is released and then a new maven-pmd-plugin 3.9.0 is released.

**Comment by Nazdravi** [comment]
After switching to Java 8 OpenJDK it is indeed possible to build PMD.
